lando/k8s/lando.py

```diff
@@ -0,0 +1,251 @@
+import logging
+import json
+import sys
+from lando.server.lando import Lando, JobStates, JobSteps, JobSettings, BaseJobActions
+from lando.k8s.cluster import ClusterApi
+from lando.k8s.jobmanager import JobManager
+from lando.k8s.config import create_server_config
+
+
+class K8sJobSettings(JobSettings):
+    def get_cluster_api(self):
+        settings = self.config.cluster_api_settings
+        return ClusterApi(settings.host, settings.token, settings.namespace, verify_ssl=settings.verify_ssl)
+
+
+class K8sJobActions(BaseJobActions):
+    """
+    Used by K8sLando to handle messages at a job specific context.
+    """
+    def __init__(self, settings):
+        super(K8sJobActions, self).__init__(settings)
+        self.cluster_api = settings.get_cluster_api()
+
+    def make_job_manager(self):
```

Comment:
It looks like all of the methods except the error handlers call `make_job_manager()` at some point. 

Sometimes it's before changing job states/steps, sometimes it's after, and sometimes it's conditional. I think we had some weird bugs in the past (#105 perhaps?) when the actions and state got out of sync. So I was wondering if there's a set of documented rules about when the states get changed or retried. What happens if `make_job_manager` raises an exception because it couldn't get the job from the API? Does the state/step get rewound?